stage0/src/handoff.rs

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
NIT: Stage0 is the only place where this half of the handoff code is used but when I was working on this initially I found it far easier to debug from a hubris task. Once I had working code in a library it was simple enough to just use the library from stage0. I'm still pretty new to rust so this may have been the wrong way to structure things? If so there's a lot of other stuff in `lib-dice` that needs to be moved down into stage0. If this is a style thing I've tended to put as much into libraries as possible and keeping this close to the consumer side of the handoff code seemed logical (to me at least).

Reply:
@flihp Great comment! I fully agree with the strategy you describe for most things. I actually had this part of the code in the `stage0-handoff` library which lives in `/lib` I moved it back into stage0 because the `stage0-handoff` library also contains the type definitions that get used on the SP and this code uses the `lpc55_pac` crate which makes it platform dependent and ends up in a symbol conflict when linking. 

I could have, I suppose, created a feature for `stage0-handoff` that only compiles this part for stage0, but that seems somewhat silly given this code only runs in stage0 ever. I also could have created two separate crates, or translation of types between the RoT and SP and kept the whole stage0-handoff crate only compiling for the lpc55, but that seemed like a touch overkill.